### ISSUE TITLE
Fix 2.0 KongClusterPlugin handling

### ISSUE THREAD

**Issue by rainest** [Kong/kubernetes-ingress-controller#1361]
As-is, 2.0 cannot see KongClusterPlugins (see https://github.com/Kong/kubernetes-ingress-controller/pull/1360#issue-654148265).

Initial review indicates that this is because the current KongClusterPlugin type is namespaced: https://github.com/Kong/kubernetes-ingress-controller/blob/e7ca5dd403f95e58dd30c7428afc4d3f76814b9f/railgun/config/crd/bases/configuration.konghq.com_kongclusterplugins.yaml#L10-L17

It should not be, as all controller code that interacts with it expects cluster-level resources by design. Debugging indicates that you can currently create namespaced KongClusterPlugins in the store fine, but the store package can't handle those.

Fixing kubebuilder tags should address this:

```
diff --git a/railgun/apis/configuration/v1/kongclusterplugin_types.go b/railgun/apis/configuration/v1/kongclusterplugin_types.go
index 90b481bf..7a5a741f 100644
--- a/railgun/apis/configuration/v1/kongclusterplugin_types.go
+++ b/railgun/apis/configuration/v1/kongclusterplugin_types.go
@@ -24,6 +24,7 @@ import (
 //+genclient
 //+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 //+kubebuilder:object:root=true
+//+kubebuilder:resource:scope=Cluster
 //+kubebuilder:subresource:status
 //+kubebuilder:storageversion
```
With that in place, [regenerating code](https://github.com/Kong/kubernetes-ingress-controller/issues/1358) does update the CRD manifest, but not the generated client. `make generate.clientsets` should, AFAIK, be the last remaining step, but it's [giving me trouble](https://gist.github.com/rainest/418d72770a43848414f0e042b8f57283).

Note that the original test from #1360 will need a small change to remove the namespace from its clientset calls once this is fixed.

**Comment by rainest** [comment]
[kcp.diff.txt](https://github.com/Kong/kubernetes-ingress-controller/files/6549820/kcp.diff.txt) is a bit more complete version of the changes with the updated test (not yet vettable due to the broken clientset signature) and post-generator CRD YAML.

**Comment by rainest** [comment]
The "unable to add directory" error in my original gist was a false alarm. When I copied the Makefile target I accidentally copied some extra whitespace that broke the command.

The target _had_ run successfully, but didn't do what I wanted because I didn't realize that kubebuilder and client-gen are both involved in this, and need different metadata tags to handle their respective tasks. Per https://github.com/Kong/kubernetes-ingress-controller/pull/1418#issuecomment-859734285 I was missing `//+genclient:nonNamespaced` originally.

**Comment by tharun208** [comment]
implemented in #1418
